tcs-service/src/main/java/com/transformuk/hee/tis/tcs/service/service/impl/ConditionsOfJoiningServiceImpl.java

```diff
@@ -43,16 +43,33 @@ public ConditionsOfJoiningServiceImpl(ConditionsOfJoiningRepository repository,
   }
 
   @Override
-  public ConditionsOfJoiningDto save(Long programmeMembershipId, ConditionsOfJoiningDto dto) {
-    LOG.info("Request received to save Conditions of Joining for Programme Membership {}.",
-        programmeMembershipId);
-
-    ProgrammeMembershipDTO programmeMembership = programmeMembershipService.findOne(
-        programmeMembershipId);
+  public ConditionsOfJoiningDto save(Object id, ConditionsOfJoiningDto dto) {
+    UUID realProgrammeMembershipId = null;
+    Long curriculumMembershipId = null;
+    ProgrammeMembershipDTO programmeMembership;
+    try {
+      realProgrammeMembershipId = UUID.fromString(id.toString());
+    } catch (IllegalArgumentException e) {
+      try {
+        curriculumMembershipId = (Long) id;
+      } catch (ClassCastException castException) {
+        throw new IllegalArgumentException(
+            String.format("%s is neither Long nor UUID", id));
+      }
+    }
+    if (realProgrammeMembershipId == null) {
+      LOG.info("Request received to save Conditions of Joining for Curriculum Membership {}.",
+          curriculumMembershipId);
+      programmeMembership = programmeMembershipService.findOne(curriculumMembershipId);
+    } else {
+      LOG.info("Request received to save Conditions of Joining for Programme Membership {}.",
+          realProgrammeMembershipId);
+      programmeMembership = programmeMembershipService.findOne(realProgrammeMembershipId);
+    }
```

Comment:
It looks like could avoid the additional `if/else` and `realProgrammeMembershipId` by including the find methods in the `try/catch`, respectively.  Don't think I'll get a chance to test that refactor until much later.

Reply:
Thanks, yes its a lot neater now!